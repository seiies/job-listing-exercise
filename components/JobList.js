import React from 'react'
import PropTypes from 'prop-types'
import JobItem from './JobItem'

const JobList = props => {
  const { jobs } = props

  return (
    <React.Fragment>
      { /* language=CSS */ }
      <style jsx>{`
            .jobs {
              margin: 0;
              padding: 0;
              list-style: none;
            }
          `}</style>

      <ul className='jobs'>
        {Array.isArray(jobs) && jobs.length
          ? jobs.map((job, idx) => (
            <JobItem
              key={idx}
              job={job}
            />
          ))
          : <p>No jobs there</p>
        }
      </ul>
    </React.Fragment>
  )
}

JobList.propTypes = {
  className: PropTypes.string,
  jobs: PropTypes.array.isRequired
}

export default JobList
