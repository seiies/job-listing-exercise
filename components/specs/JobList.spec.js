/* global describe, it, expect */

import React from 'react'
import JobList from '../JobList'
import { shallow } from 'enzyme'

describe('JobList', () => {
  let jobs = [
    {
      id: 1,
      image: 'https://place.cat/c/100/100',
      company: 'Deutsche Bahn AG',
      title: 'Job title'
    },
    {
      id: 2,
      image: 'https://place.cat/c/100/100',
      company: 'Deutsche Bahn AG',
      title: 'Job title'
    }
  ]
  let noJobs = []

  it('renders correctly with job list', () => {
    const component = shallow(<JobList jobs={jobs} />)
    expect(component).toMatchSnapshot()
  })

  it('renders correctly with empty job list', () => {
    const component = shallow(<JobList jobs={noJobs} />)
    expect(component).toMatchSnapshot()
  })
})
