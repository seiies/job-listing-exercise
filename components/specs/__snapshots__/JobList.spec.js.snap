// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobList renders correctly with empty job list 1`] = `
<React.Fragment>
  <JSXStyle
    css=".jobs.jsx-3773623910{margin:0;padding:0;list-style:none;}"
    styleId="3773623910"
  />
  <ul
    className="jsx-3773623910 jobs"
  >
    <p
      className="jsx-3773623910"
    >
      No jobs there
    </p>
  </ul>
</React.Fragment>
`;

exports[`JobList renders correctly with job list 1`] = `
<React.Fragment>
  <JSXStyle
    css=".jobs.jsx-3773623910{margin:0;padding:0;list-style:none;}"
    styleId="3773623910"
  />
  <ul
    className="jsx-3773623910 jobs"
  >
    <JobItem
      job={
        Object {
          "company": "Deutsche Bahn AG",
          "id": 1,
          "image": "https://place.cat/c/100/100",
          "title": "Job title",
        }
      }
      key="0"
    />
    <JobItem
      job={
        Object {
          "company": "Deutsche Bahn AG",
          "id": 2,
          "image": "https://place.cat/c/100/100",
          "title": "Job title",
        }
      }
      key="1"
    />
  </ul>
</React.Fragment>
`;
